In [228]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.grid_search import GridSearchCV
from sklearn.linear_model import Lasso
from sklearn.linear_model import Ridge

In [229]:
data = pd.read_csv('YearlyMovieGross2000to20144.csv')

In [230]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8462 entries, 0 to 8461
Data columns (total 15 columns):
Year                     8462 non-null int64
Rank                     8462 non-null int64
Movie Title              8462 non-null object
Studio                   8436 non-null object
Total Gross              8462 non-null int64
Theaters                 8390 non-null float64
Opening                  8173 non-null float64
Opening Theaters Open    8215 non-null float64
Gross.Theater            8459 non-null float64
2014 dollars             8462 non-null int64
Length Weeks             8459 non-null float64
Number 1 Weeks           8462 non-null int64
Gross per week           8174 non-null float64
Fiscal Year              8176 non-null float64
Quarter Fiscal Year      8176 non-null float64
dtypes: float64(8), int64(5), object(2)
memory usage: 1.0+ MB


In [231]:
data = data.dropna()

In [232]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7951 entries, 0 to 8461
Data columns (total 15 columns):
Year                     7951 non-null int64
Rank                     7951 non-null int64
Movie Title              7951 non-null object
Studio                   7951 non-null object
Total Gross              7951 non-null int64
Theaters                 7951 non-null float64
Opening                  7951 non-null float64
Opening Theaters Open    7951 non-null float64
Gross.Theater            7951 non-null float64
2014 dollars             7951 non-null int64
Length Weeks             7951 non-null float64
Number 1 Weeks           7951 non-null int64
Gross per week           7951 non-null float64
Fiscal Year              7951 non-null float64
Quarter Fiscal Year      7951 non-null float64
dtypes: float64(8), int64(5), object(2)
memory usage: 993.9+ KB


In [233]:
# set our training
y, X = data['Total Gross'], data[['Opening','Theaters']]

In [234]:
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=1)

In [235]:
# set our training to include Rank, Opening Theaters and Gross per Theater
y, X = data['Total Gross'], data[['Opening', 'Theaters', 'Length Weeks']]

from sklearn.preprocessing import StandardScaler
stdsc = StandardScaler()
X_train_std = stdsc.fit_transform(X_train)
X_test_std = stdsc.transform(X_test)
X_train_std.shape

from sklearn.cross_validation import ShuffleSplit

from sklearn.grid_search import GridSearchCV

from sklearn.cross_validation import ShuffleSplit
cv = ShuffleSplit(X_train.shape[0], n_iter=10, test_size=0.2, random_state=0)

from sklearn.linear_model import LinearRegression
lr = LinearRegression()
lr.fit(X_train_std,y_train)

lr.score(X_test_std,y_test)

0.85687610521945889

In [236]:
#when I ran it just using X and y values, my score was better.

lr.fit(X,y)

lr.score(X,y)

0.86490432473418077

In [237]:
#when I ran it just using X and y values, my score was better.

lr.fit(X,y)

lr.score(X,y)

0.86490432473418077

In [238]:
# set our training to include Year as well 
y, X = data['Total Gross'], data[['Opening','Theaters', 'Length Weeks', 'Year']]

In [239]:
lr.fit(X_train_std,y_train)

lr.score(X_test_std,y_test)

0.85687610521945889

In [240]:
#when I ran it just using X and y values, my score was better.

lr.fit(X,y)

lr.score(X,y)

0.86508987470491339

In [241]:
# set our training to include Year as well 
y, X = data['Total Gross'], data[['Opening','Theaters', 'Year','Gross per week', 'Number 1 Weeks', 'Length Weeks']]

In [242]:
#my model became even more accurate when I added in Gross per week and number 1 weeks
lr = LinearRegression()
lr.fit(X_train_std,y_train)

lr.score(X_test_std,y_test)

0.85687610521945889

In [243]:
#when I ran it just using X and y values, my score was better.

lr.fit(X,y)

lr.score(X,y)

0.89915957898616317

In [244]:
# set our training to include Rank, Opening Theaters and Gross per Theater
y, X = data['Total Gross'], data[['Opening','Theaters', 'Year','Gross per week', 'Number 1 Weeks', 'Length Weeks', 
                                  'Rank','Opening Theaters Open', 'Gross.Theater']]

In [245]:
#when I ran it just using X and y values, my score was better.

lr.fit(X,y)

lr.score(X,y)

0.91818102823677117

In [246]:
lr = LinearRegression()
lr.fit(X_train_std,y_train)

lr.score(X_test_std,y_test)

0.85687610521945889

My score was 92 percent accurate using a lot of the feautres from my dataset and X and y values.

In [247]:
lr.coef_

array([ 36266648.60380022,   6167972.51948565])

Opening Week Gross was 6 times more important than amount of Theaters the movie opened in, in terms of predicting total gross of a movie.

In [248]:
# create a dataframe to show our coefficients
d = {'features': X_train.columns, 'coefficients': lr.coef_}
pd.DataFrame(d)

coefficients  features
0  36266648.603800   Opening
1   6167972.519486  Theaters

In [249]:
# Linear Regression with lasso

from sklearn.grid_search import GridSearchCV
from sklearn.linear_model import Lasso
parameters = {'alpha': np.linspace(1,10,20)}
lasso = Lasso(normalize=True)

In [250]:
print(lr.intercept_)

17951241.4109


When I used my x_train and y_train set, my model was only 86% accurate. My score did not change, even when I added in new features. When I used X and Y, my model was 92% accurate. Again, Opening Week Gross was 6 times more important than the number of theaters a movie opened in!